docs/discussion/state-management.md

```diff
@@ -72,66 +72,25 @@ export function List() {
 }
 ```
 
-Now consider you want the URL to update when the user changes the view. Note the state synchronization:
-
-```tsx bad lines=[10,19,27]
-import {
-  useNavigate,
-  useSearchParams,
-} from "@remix-run/react";
-
-export function List() {
-  const navigate = useNavigate();
-  const [searchParams] = useSearchParams();
-  const [view, setView] = React.useState(
-    searchParams.get("view") || "list"
-  );
-
-  return (
-    <div>
-      <div>
-        <button
-          onClick={() => {
-            setView("list");
-            navigate(`?view=list`);
-          }}
-        >
-          View as List
-        </button>
-        <button
-          onClick={() => {
-            setView("details");
-            navigate(`?view=details`);
-          }}
-        >
-          View with Details
-        </button>
-      </div>
-      {view === "list" ? <ListView /> : <DetailView />}
-    </div>
-  );
-}
-```
-
-Instead of synchronizing state, you can simply read and set the state in the URL directly with boring old HTML forms.
+Instead of using state, you could just use links and search parameters.
```

Comment:
I think you cut out too much here. The point is still that you can "simply read and set the state in the URL directly"